pkg/tsdb/cloudwatch/models/cloudwatch_query.go

```diff
@@ -27,13 +27,36 @@ type cloudWatchQuery struct {
 	MatchExact        bool
 	UsedExpression    string
 	TimezoneUTCOffset string
-	MetricQueryType   metricQueryType
-	MetricEditorMode  metricEditorMode
+	MetricQueryType   MetricQueryType
+	MetricEditorMode  MetricEditorMode
 }
 
-func (q *cloudWatchQuery) getGMDAPIMode() gmdApiMode {
+type (
+	MetricQueryType  uint32
+	MetricEditorMode uint32
+	GMDApiMode       uint32
+)
+
+const (
+	MetricEditorModeBuilder MetricEditorMode = iota
+	MetricEditorModeRaw
+)
+
+const (
+	MetricQueryTypeSearch MetricQueryType = iota
+	MetricQueryTypeQuery
+)
+
+const (
+	GMDApiModeMetricStat GMDApiMode = iota
+	GMDApiModeInferredSearchExpression
+	GMDApiModeMathExpression
+	GMDApiModeSQLExpression
+)
+
+func (q *CloudWatchQuery) GetGMDAPIMode() GMDApiMode {
```

Comment:
I put these constants and the deeplink types and functions below in this file for now as a conservative approach (as opposed to other file(s) with these on their own). May become clearer where they belong in a subequent refactor.